File: problems/1624-largest-substring-between-two-equal-characters/solution.cpp
class Solution {
public:
    int maxLengthBetweenEqualCharacters(string s) {
        int maxLength = -1;
        for (int i = 0; i < s.length(); ++i) {
            for (int j = s.length() - 1; j > i; --j) {
                if (s[i] == s[j]) {
                    maxLength = max(maxLength, j - i - 1);
                    break;
                }
            }
        }
        return maxLength;
    }
};

File: problems/1624-largest-substring-between-two-equal-characters/solution.java
class Solution {
    public int maxLengthBetweenEqualCharacters(String s) {
        int maxLength = -1;
        for (int i = 0; i < s.length(); i++) {
            for (int j = s.length() - 1; j > i; j--) {
                if (s.charAt(i) == s.charAt(j)) {
                    maxLength = Math.max(maxLength, j - i - 1);
                    break;
                }
            }
        }
        return maxLength;
    }
}